servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcRouter.java

```diff
@@ -572,7 +574,18 @@ public void closeGracefully() throws Exception {
                         @Override
                         public void handle(final GrpcServiceContext ctx, final BlockingIterable<Req> request,
                                            final GrpcPayloadWriter<Resp> responseWriter) throws Exception {
-                            route.handle(ctx, requireNonNull(request.iterator().next()), responseWriter);
+                            final Req firstItem;
+                            try (BlockingIterator<Req> requestIterator = request.iterator()) {
+                                firstItem = requireNonNull(requestIterator.next(), "Request item is null");
```

Comment:
It isn't changed in this PR but I think `null` should pass-through.

Reply:
Our API for gRPC currently does not annotate client calls with `@Nullable` annotation. If I don't do a null check, intellij idea warns that we return a possible `null` value from `@Nonnull` method.
And the `ProtoBufSerializationProvider` never returns null, even if the `Buffer` was empty: https://github.com/apple/servicetalk/blob/c4ab8877ace59689f67af2f4e4ad8e9bcb1939e1/servicetalk-grpc-protobuf/src/main/java/io/servicetalk/grpc/protobuf/ProtoBufSerializationProvider.java#L121-L123
Therefore, it never will be `null`. Only `NoSuchElementException` is possible here.

Btw, on the client-side we can simply throw `NoSuchElementException`. But on the server-side, do we need to generate a `GrpcStatusException` with `INVALID_ARGUMENT` if there were no items in the request?

Reply:
>> Therefore, it will never be null. Only NoSuchElementException is possible here.

Can a filter modify the request payload to pass `null` since we allow `null` for our async sources?
The meta point here is that unless we require the item to be `non-null`, we should allow for `null` values to pass through as we do in all our async sources. If the intent is only to avoid code style/spotbugs warnings then we can ignore/suppress.

Reply:
Emitting a `GrpcStatusException` SGTM